### PR TITLE
Implement yaml files for training runs from Garg et al.

### PR DESCRIPTION
**Pull request by Famara72**

### CODE REVIEW

**Review by nelson-lojo — CHANGES_REQUESTED** [review]
Can you correct the model configs for GPT2, Llama, and all of the baselines? You can name the baselines by their name and the models by `std_<gpt2/llama>` 

In the same vein, can you make write the configs for all **4** of the training runs?

File: conf/models.yml [hunk 1]
```diff
@@ -21,3 +21,12 @@ model: &least_squares
...
     type: xgboost
+
+model: &averaging
+    type: averaging
+
+model: &decision_tree
+    type: decision_tree
+
+model: &mlp
+    type: mlp
```
Comment (nelson-lojo) [inline conf/models.yml:32]: please end yaml files with a newline -- it makes it easier to handle when combining files programmatically

File: conf/models.yml
Comment (nelson-lojo) [inline conf/models.yml:1]: * are the configrations for llama and gpt2 correct?
* I don't see lasso here but it's referenced in the sparse_linear training runs
* The zero model isn't here ... I understand it's not really sensible to use during training, but it's probably best to have it available in this file

**Review by nelson-lojo — CHANGES_REQUESTED** [review]
Minor changes needed to wrap this up

File: conf/models.yml (hunk 1 above)
Comment (Famara72) [reply conf/models.yml:32]: Ok, will do

File: conf/models.yml
Comment (Famara72) [reply conf/models.yml:1]: Done

File: conf/models.yml [hunk 2]
```diff
@@ -21,3 +21,18 @@ model: &least_squares
...
```
Comment (nelson-lojo) [inline conf/models.yml:32]: we should sync this up with [what's in Garg et al.](https://github.com/dtsip/in-context-learning/blob/main/src/models.py#L51-L64) the defaults are much smaller than that to make CI run faster

**Review by nelson-lojo — CHANGES_REQUESTED** [review]
We missed the [baseline configs from Garg et al.](https://github.com/dtsip/in-context-learning/blob/main/src/models.py#L29-L74)

File: conf/models.yml [hunk 3]
```diff
@@ -21,3 +21,43 @@ model: &least_squares
...
+    max_depth: 4
+
+model: &decision_tree
+    type: decision tree
+    max_depth: None
+
+
+model: &mlp
+    type: grad mlp
+    dimensions: [20, 100, 1]
```
Comment (nelson-lojo) [inline conf/models.yml]: This should be [100], The in dimension and out dimension are filled in by the parsing code

File: conf/train/decision_tree_gpt2.yml [hunk 4]
```diff
@@ -22,4 +23,4 @@ steps: 500000
 log_freq: 100
 checkpoint_freq: 5000
 model: *std_gpt2
-baseline_models: [*least_squares, *knn_3, *xgboost]
+baseline_models: [*knn_3, *xgboost, *decision_tree, *decision_tree_4]
```
Comment (nelson-lojo) [inline conf/train/decision_tree_gpt2.yml:26]: Does `max_depth=None` for the decision tree let it "fit" until all points have a leaf? I'm not super familiar with sklearn in this regard

File: conf/train/test.yml
Comment (nelson-lojo) [inline conf/train/test.yml:1]: don't check this file into git please. It was meant to be a local check

**Review by nelson-lojo — CHANGES_REQUESTED** [review]
minor changes

File: conf/models.yml (hunk 3 above)
Comment (nelson-lojo) [reply conf/models.yml]: It says outdated because this ended up nested in another argument, but the change is still necessary

File: conf/train/test.yml
Comment (Famara72) [reply conf/train/test.yml:1]: removed

File: conf/train/decision_tree_gpt2.yml (hunk 4 above)
Comment (Famara72) [reply conf/train/decision_tree_gpt2.yml:26]: yeah or until you have less than the minimum required for a split

**Review by nelson-lojo — APPROVED** [review]
Nice! Looks good to me!

File: conf/models.yml (hunk 3 above)
Comment (nelson-lojo) [reply conf/models.yml]: fixed.

**Pull request merged**